pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/BKCluster.java

```diff
@@ -231,9 +234,29 @@ private ServerConfiguration newServerConfiguration(int index) throws Exception {
             // and 2nd bookie's cookie validation fails
             port = clusterConf.bkPort;
         }
+        File[] cookieDir = dataDir.listFiles((file) -> file.getName().equals("current"));
+        if (cookieDir != null && cookieDir.length > 0) {
+            String existBookieAddr = parseBookieAddressFromeCookie(cookieDir[0]);
+            if (existBookieAddr != null) {
+                baseConf.setAdvertisedAddress(existBookieAddr.split(":")[0]);
+                port = Integer.parseInt(existBookieAddr.split(":")[1]);
+            }
+        }
         return newServerConfiguration(port, dataDir, new File[]{dataDir});
     }
 
+    private String parseBookieAddressFromeCookie(File dir) {
+        Cookie cookie;
+        try {
+            cookie = Cookie.readFromDirectory(dir);
+        } catch (IOException e) {
```

Comment:
We can just throw out the exception. 
Otherwise, we will not able to know the details of the IOException

Reply:
Thanks for your advice, I'll fix it.